react-delegationdashboard/src/components/NodesList/index.tsx

```diff
@@ -65,21 +90,43 @@ const NodesTable = () => {
                 status = { key: value.asString, value: NodeStatus[value.asString] };
             }
             else {
+                if (status.key === 'queued') {
+                    queued.push(value.asHex);
+                }
                 nodes.push(new NodeType(value.asHex, status));
             }
         });
     };
 
     const getDiplayNodes = () => {
         Promise.all([getAllNodesStatus(), getBlsKeysStatus()]).then(
-            ([nodesStatus, blsKeys]) => {
+            async ([nodesStatus, blsKeys]) => {
                 let temp = nodesStatus;
                 temp.map(item => {
```

Comment:
This code is hard to follow, maybe split in a function to describe why the status from one array is moved to the other?

Reply:
And maybe a better use of `map` is to:
```
const adaptedNodesStatus = nodesStatus.map(item => {
  let index = blsKeys.findIndex(i => i.blsKey === item.blsKey);
  return {
    ...item,
    status: index >= 0 ? blsKeys[index].status : item.status
  };
})
```